includes/admin/tools/data/class-give-tools-delete-test-transactions.php

```diff
@@ -65,61 +63,10 @@ public function get_data() {
 		$step_items = array_slice( $items, $offset, $this->per_step );
 
 		if ( $step_items ) {
-
-			$step_ids = array(
-				'other' => array(),
-			);
-
 			foreach ( $step_items as $item ) {
-
-				$step_ids['other'][] = $item['id'];
-
-			}
-
-			$sql = array();
-
-			foreach ( $step_ids as $type => $ids ) {
-
-				if ( empty( $ids ) ) {
-					continue;
-				}
-
-				$parent_query = '';
-
-				switch ( $type ) {
-					case 'other':
-
-						$temp_ids = implode( ',', $ids );
-
-						// Get all the test logs of the donations ids.
-						$parent_query = "SELECT DISTINCT post_id as id FROM $wpdb->postmeta WHERE meta_key = '_give_log_payment_id' AND meta_value IN ( $temp_ids )";
-						$parent_ids   = $wpdb->get_results( $parent_query, 'ARRAY_A' );
-
-						// List of all test logs.
-						if ( $parent_ids ) {
-							foreach ( $parent_ids as $parent_id ) {
-								// Adding all the test log in post ids that are going to get deleted.
-								$ids[] = $parent_id['id'];
-							}
-						}
-						$ids = implode( ',', $ids );
-
-						$sql[] = "DELETE FROM $wpdb->posts WHERE id IN ($ids)";
-						$sql[] = "DELETE FROM $wpdb->postmeta WHERE post_id IN ($ids)";
-						$sql[] = "DELETE FROM $wpdb->comments WHERE comment_post_ID IN ($ids)";
-						$sql[] = "DELETE FROM $wpdb->commentmeta WHERE comment_id NOT IN (SELECT comment_ID FROM $wpdb->comments)";
-						break;
-				}
-
+				// Delete the main payment.
+				give_delete_donation( (int) $item['id'] );
```

Comment:
There's no fear of this breaking down with large site with lots of test donations? I can test with our GiveWP site. Have you tested @raftaar1191 ?

Reply:
Yes I have Tested It Completely @DevinWalker

Reply:
@raftaar1191 Please use [`absint`](https://codex.wordpress.org/Function_Reference/absint)

Reply:
Done Thanks @ravinderk to pointing this out